### PR TITLE
Enhancements/stats docs

### PR DESCRIPTION
**Pull request by kieranricardo**
This PR adds some documentation for dataset fusing, caching data from multiple products, and generating mosaics.

It also makes mosaics API a little smoother by allowing more flexible for specify input datasets docs.

### CODE REVIEW

File: libs/stats/README.md [hunk 1]
```diff
@@ -277,3 +292,12 @@ You can now run a query on the log of the selected job:
 
 <img src="docs/auxiliary/screenshot-grafana.png" alt="Screenshot Grafana" width="1000"/>
 
+## Mosaics
+
+To quickly review products stats can be used to generate a low resolution continental mosaic of a summary product. To run the command:
+
+```
+odc-stats generate-mosaic product_definition.yaml s3://input-data-bucket/folder/*/*/metadata-file.stac-item.json s3://output-bucket/folder/subfolder --bands nbart_red,nbart_blue
```
Comment (omad) [inline libs/stats/README.md]: The example glob should have quote characters around it.

What goes into the `product_definition.yaml`?
I assume the output is a TIFF file for each band, but this should be stated.

Is the output actually a COG? How many overview layers are generated? Can the resampling method be specified? 

Does this/How does this parallelise across CPUs? What's the expected runtime and memory usage when running this?

**Review by omad — CHANGES_REQUESTED** [review]
Nice improvements Kieran!

Can you please include some more information about the mosaic generation please.

File: libs/stats/README.md (hunk 1 above)
Comment (kieranricardo) [reply libs/stats/README.md]: I'll add those quotes in :) 

The product definition is just a standard datacube product definition file. This enables the mosaicing to be done for stuff that isn't indexed. I'll make this a bit cleearer.

Yep its a proper COG! It has 7 overview layers (maybe fewer for smaller images?) And the resampling can't be specified right now but I can add this in.

It's fully parallelised, hopefully run time is `~1/num_cores`. Not exactly sure what the memory usage it's but probably something like `2048 * 2048 * bands * dtype_size * num_cores `, so for 64 cores maybe something like ~1GB ish. I think? Like each core gets a `2048 * 2048` chunk of the whole continent.

**Review by omad — APPROVED** [review]

**Comment by omad** [comment]
Thanks Kieran. I've rebased and cleaned up the commits (generally appreciated) and added your comments about runtime resource requirements.

**Pull request merged**